File: shopping-list-gestor-service/src/main/java/es/franricodev/shopping_list_gestor_service/shoppinglist/dto/response/ResponseGetShoppinglistDetailsMetadata.java
package es.franricodev.shopping_list_gestor_service.shoppinglist.dto.response;

import es.franricodev.shopping_list_gestor_service.shoppinglistitem.ShoppinglistItemMetadataDTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.util.List;

@Getter
@Setter
@ToString
@Builder
public class ResponseGetShoppinglistDetailsMetadata {
    private ShoppinglistMetadata shoppinglistMetadata;
    private List<ShoppinglistItemMetadataDTO> items;
}

File: shoppinglist-gestor-mobile/src/Shoppinglist/infrastructure/models/ResponseGetShoppinglistDetailsMetadata.ts
import { ShoppinglistMetadata } from "@/Shoppinglist/domain/ShoppinglistMetadata";
import { ShoppinglistItemMetadata } from "@/ShoppinglistItem/domain/ShoppinglistItemMetadata";

export interface ResponseGetShoppinglistDetailsMetadata {
    shoppinglistMetadata: ShoppinglistMetadata
    items: ShoppinglistItemMetadata[]
}